scripts/execute-command.ts

```diff
@@ -0,0 +1,414 @@
+    } else {
+      secretName = `vargasjr-prod-prod-key-pem`;
+    }
+
+    console.log(`🔑 Retrieving SSH key from secret: ${secretName}`);
+    
+    try {
+      return await getSecret(secretName, this.config.region);
+    } catch (error: any) {
+      if (this.config.prNumber && error.message?.includes('Secret not found')) {
+        const fallbackSecretName = `vargasjr-pr-${this.config.prNumber}-key-pem`;
+        console.log(`Primary secret not found, trying fallback: ${fallbackSecretName}`);
+        return await getSecret(fallbackSecretName, this.config.region);
+      }
+      throw error;
+    }
+  }
+
+  private async writeKeyToTempFile(keyMaterial: string, keyName: string): Promise<string> {
+    const keyPath = `${tmpdir()}/vargasjr-ssh-${keyName}-${Date.now()}.pem`;
+    writeFileSync(keyPath, keyMaterial, { mode: 0o600 });
+    console.log(`✅ SSH key written to temporary file: ${keyPath}`);
+    return keyPath;
+  }
+
+  private displayResults(sshResults: any, ssmResults: any): void {
+    console.log(`\n📊 EXECUTION SUMMARY`);
+    console.log(`==================`);
+    
+    console.log(`\n🔗 SSH Results:`);
+    if (sshResults.success) {
+      const successCount = sshResults.results.filter((r: any) => r.success).length;
+      const failCount = sshResults.results.length - successCount;
+      console.log(`   ✅ ${successCount} successful, ❌ ${failCount} failed`);
+      
+      sshResults.results.forEach((result: any) => {
+        const status = result.success ? '✅' : '❌';
+        console.log(`   ${status} ${result.name} (${result.instanceId})`);
+        if (result.error) {
+          console.log(`      Error: ${result.error}`);
+        }
+      });
+    } else {
+      console.log(`   ❌ Failed: ${sshResults.error}`);
+    }
+    
+    console.log(`\n📡 SSM Results:`);
+    if (ssmResults.success) {
+      const successCount = ssmResults.results.filter((r: any) => r.success).length;
+      const failCount = ssmResults.results.length - successCount;
+      console.log(`   ✅ ${successCount} successful, ❌ ${failCount} failed`);
+      
+      ssmResults.results.forEach((result: any) => {
+        const status = result.success ? '✅' : '❌';
+        console.log(`   ${status} ${result.name} (${result.instanceId})`);
+        if (result.error) {
+          console.log(`      Error: ${result.error}`);
+        }
+      });
+    } else {
+      console.log(`   ❌ Failed: ${ssmResults.error}`);
+    }
+  }
+}
+
+async function main() {
+  const command = process.env.COMMAND;
+  if (!command) {
+    console.error("❌ COMMAND environment variable is required");
+    process.exit(1);
+  }
+
+  let prNumber: string | undefined;
+  const githubRef = process.env.GITHUB_REF;
+  if (githubRef) {
+    const prMatch = githubRef.match(/refs\/heads\/.*pr[/-](\d+)/i);
+    if (prMatch) {
+      prNumber = prMatch[1];
+      console.log(`🔍 Inferred PR number from branch: ${prNumber}`);
+    }
+  }
```

Comment:
This gets the branch name - we'll need to use that to then infer or number. Look at our one time migration runner for how we use the GitHub api to resolve this logic, and consolidate into a utility if possible